internal/keppel/rbac_policy.go

```diff
@@ -20,16 +20,113 @@
 package keppel
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+
 	"github.com/sapcc/go-bits/regexpext"
 )
 
-// NewRBACPolicy is a policy granting user-defined access to repos in an account.
+// RBACPolicy is a policy granting user-defined access to repos in an account.
 // It is stored in serialized form in the RBACPoliciesJSON field of type Account.
-//
-// TODO: rename to type RBACPolicy after the `rbac_policies` table has been removed
-type NewRBACPolicy struct {
-	CidrPattern       string                `json:"match_cidr,omitempty"`
-	RepositoryPattern regexpext.PlainRegexp `json:"match_repository,omitempty"`
-	UserNamePattern   regexpext.PlainRegexp `json:"match_username,omitempty"`
-	Permissions       []string              `json:"permissions"`
+type RBACPolicy struct {
+	CidrPattern       string                  `json:"match_cidr,omitempty"`
+	RepositoryPattern regexpext.BoundedRegexp `json:"match_repository,omitempty"`
+	UserNamePattern   regexpext.BoundedRegexp `json:"match_username,omitempty"`
+	Permissions       []RBACPermission        `json:"permissions"`
+}
```

Comment:
FYI: The use of `PlainRegexp` was copied from `internal/api/keppel/accounts.go`, but turned out to be incorrect because the match patterns are defined as implying `^` and `$` anchors, i.e. the BoundedRegexp behavior. This was not a problem before because the old type in the API package only did input validation, not actual matching.